server/src/test/java/keywhiz/service/providers/ClientAuthFactoryTest.java

```diff
@@ -193,37 +206,71 @@ public void clientWhenPrincipalAbsentThrows() {
     assertThat(factory.provide(request, httpServletRequest)).isEqualTo(client);
   }
 
-  @Test(expected = NotAuthorizedException.class)
-  public void rejectsXfcc_requesterAuthMissing() {
+  @Test public void returnsClientWhenClientPresent_fromXfccHeader_customSpiffeIdHeader() {
```

Comment:
Do we ever expect both the XFCC header and the custom SPIFFE ID header to be set? If so, Keywhiz should check that the same client is extracted from both headers. If not, Keywhiz should check that only one header is set and deny access if they're both set, since the request is malformed.